patches/minecraft/net/minecraft/world/item/BucketItem.java.patch

```diff
@@ -47,6 +47,16 @@
                 if (this.m_142073_(p_40704_, p_40703_, blockpos2, blockhitresult)) {
                    this.m_142131_(p_40704_, p_40703_, itemstack, blockpos2);
                    if (p_40704_ instanceof ServerPlayer) {
+@@ -109,6 +_,9 @@
+          boolean flag1 = blockstate.m_60795_() || flag || block instanceof LiquidBlockContainer && ((LiquidBlockContainer)block).m_6044_(p_150717_, p_150718_, blockstate, this.f_40687_);
+          if (!flag1) {
+             return p_150719_ != null && this.m_142073_(p_150716_, p_150717_, p_150719_.m_82425_().m_121945_(p_150719_.m_82434_()), (BlockHitResult)null);
++         } else if (this.f_40687_.getFluidType().isVaporizedOnPlacement(p_150717_, p_150718_, new net.minecraftforge.fluids.FluidStack(this.f_40687_, net.minecraftforge.fluids.FluidType.BUCKET_VOLUME))) {
++            this.f_40687_.getFluidType().onVaporize(p_150716_, p_150717_, p_150718_, new net.minecraftforge.fluids.FluidStack(this.f_40687_, net.minecraftforge.fluids.FluidType.BUCKET_VOLUME));
```

Comment:
Don't create a raw fluid stack, grab it from the capability on the bucket.

Reply:
Unfortunately, there is no access to the involved itemstack or the players hand in the `BucketItem::emptyContents` method  and I don't want to change too much there.

Reply:
@ChampionAsh5357 What should I do with this?

Reply:
Sorry for the late response, we should probably expand the method to take in the ItemStack and then deprecate the old one.

Reply:
Thanks for your answer. I am working on that and I noticed, that this method is also used in the vanilla interface `DispensibleContainerItem`. It is called by the dispenser and the `SolidBucketItem` implements that interface, too. Should I change also this interface and add the method with the item stack?

Reply:
Just redirect the call to the one with the itemstack yeah.

Reply:
I hope the actual commit conatins the changes you described here. :)